### ISSUE TITLE
Read and visualize STAC GeoParquet (via duckdb WASM?)

### ISSUE THREAD

**Issue by m-mohr** [radiantearth/stac-browser#493]
As an alternative datasource, we could also read from files that comply to https://github.com/stac-utils/stac-geoparquet
It would need a geoparquet parser, e.g. hyparquet, parquet-wasm, duckdb WASM, ...

Anyone interested in this?

**Comment by betolink** [comment]
This would be really neat, we've been prototyping serverless STAC and the missing component is precisely this.

**Comment by santilland** [comment]
Absolutely, also working with serverless STAC and we see a lot of benefits using geoparquets. Only thing holding us a bit back is that it is not natively supported by other components, such as STAC Browser

**Comment by jo-chemla** [comment]
Interesting take by @carbonplan on a wasm geoparquet reader, repo available here [carbonplan/geoparquet-web](https://github.com/carbonplan/geoparquet-web) and [webapp](https://geoparquet-web.vercel.app), presented at Cloud Native Geospatial 2025 + [talk video recording](https://www.youtube.com/watch?v=SsjNPFv5dgI)

**Comment by m-mohr** [comment]
Some experiments with <https://demo.duckui.com>:

On my 6 yr old desktop on a ~60 Mbit/s connection I tried to load <https://data.source.coop/cholmes/stac-geoparquet-public/slim/landsat-stac.parquet>. (Note: It has no assets and links.)

- geometry column is pretty large in comparison to the rest of the data, so should not be loaded by default
- duchdb loads the chunks sequentially, thus is rather slow.
- For an overview of items, we'd probably want to load only some minimal information that's needed for the card view, e.g. `SELECT id, datetime, thumbnail_url FROM 'https://data.source.coop/cholmes/stac-geoparquet-public/slim/landsat-stac.parquet' LIMIT 20;` - This transferred ~4MB in ~11 seconds. So STAC Browser would be rather slow with expected wait times > 10 seconds even in normal and rather simple usecases.
- Ideally, we would also load e.g. start_datetime, end_datetime, title, maybe bbox. These are often not available, so we should always start with a `DESCRIBE 'https://data.source.coop/cholmes/stac-geoparquet-public/slim/landsat-stac.parquet'` (which is returning really fast)
- Potentially interesting could be `SELECT * FROM parquet_metadata('https://data.source.coop/cholmes/stac-geoparquet-public/slim/landsat-stac.parquet');` and `SELECT * FROM parquet_schema('https://data.source.coop/cholmes/stac-geoparquet-public/slim/landsat-stac.parquet');`
- Couldn't find a way to extrect collections from the headers, but not sure whether they are actually included. There's `key_value_metadata` in the columns that `parquet_metadata` returns, but only for the columns, not for the whole file.
- Requesting a single item (from the beginning, so ideal case) through `SELECT * FROM 'https://data.source.coop/cholmes/stac-geoparquet-public/slim/landsat-stac.parquet' WHERE id = 'LT04_L2SP_005027_19820824_02_T1' LIMIT 1` transferred ~33MB in ~43 secs. (Requesting a non-existing ID is at least rather fast, < 1 sec.)
- Querying isn't actually really slow. `SELECT id, datetime, thumbnail_url, "eo:cloud_cover" FROM 'https://data.source.coop/cholmes/stac-geoparquet-public/slim/landsat-stac.parquet' WHERE "eo:cloud_cover" < 10 LIMIT 20` is similarly fast as just running the same command without WHERE clause. I didn't test spatial or temporal queries yet.

Conclusion: Unless you provide a highly optimized GeoParquet (e.g. small row group sizes), the "normal" STAC Browser experience will be pretty slow.
